src/airflow/dags/dag_preprocess.py

```diff
@@ -0,0 +1,27 @@
+"""Airflow DAG for the Preprocess part of the pipeline."""
+from __future__ import annotations
+
+from pathlib import Path
+
+from airflow.models.dag import DAG
+
+from . import common_airflow as common
+
+CLUSTER_NAME = "otg-preprocess"
+
+ALL_STEPS = [
```

Comment:
I see what you mean; but as it turns out, reading configuration from YAML is not a good idea in Airflow, in general, for two reasons.

**1. Conceptual.** The _foundation_ of Airflow is "configuration as code": [\[1\]](https://airflow.apache.org/docs/apache-airflow/2.7.1/#workflows-as-code), [\[2\]](https://big-data-test-infrastructure.ec.europa.eu/service-offering/apache-airflow_en). The idea is, because Python is more extensible, expressive and elegant than YAML or JSON, it just makes sense to define all concepts directly, rather reading them from external files.

**2. Technical.** As we discussed previously with Manuel and David, to ensure that the DAG is current, Airflow scans and re-interprets the Python file multiple times per second. Because of that, it is important to make each DAG file as lightweight as possible on importing external libraries and parsing files. And, well, defining a static list of several strings in Python is about _6,700 times faster_ than reading it from an external YAML file (note the change in units between the approaches):

```python
%%timeit -n 1000
steps = ["finngen", "eqtl_catalog", "gwas_catalog", "uk_biobank"]
```
**29.7 ns** ± 0.246 ns per loop (mean ± std. dev. of 7 runs, 1000 loops each)

```python
%%timeit -n 1000
from pathlib import Path
import yaml
config_path = Path("test.yaml")
assert config_path.exists(), f"YAML config path {config_path} does not exist."
with open(config_path, "r") as config_file:
    steps = list(yaml.safe_load(config_file))
```
**201 µs** ± 2.61 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

So actually, a good idea would be to eventually move away from using YAML in the ETL DAG as well; this would make the entire setup faster and more expressive at the same time

Reply:
Yes, certainly it is possible to use YAMLs and in some cases even inevitable — for example, if we don't know the structure of the workflow in advance and it is provided externally. Our workflows are, however, always predefined (at least at this point in the development).

As for this part:

> a system that a non expert can tweak

I think that this would be fantastic (really!), but at the same time, given the complexity of the entire project, I think it's pretty impossible :-)